CHANGELOG.md

```diff
@@ -1,5 +1,55 @@
 # Changelog
 
+## 0.4.0
+
+This release adds first-class support for Python projects that are not designed as Python packages (e.g., web applications, data science projects, etc.).
+
+In doing so, it also includes some breaking changes around uv's handling of such projects. Previously, uv required that all projects (with the minor exception of virtual workspace roots) were buildable, and always installed such projects into the virtual environment. Projects created by `uv init` always included a `[build-system]` definition and existing projects that did not define a `[build-system]` would use the legacy setuptools build backend by default.
+
+However, most users are not developing libraries that need to be packaged and published to PyPI. Instead, they're building applications using web frameworks, or running collections of Python scripts in the project's root directory. In these cases, requiring a `[build-system]` was confusing and error-prone. In this release, uv changes the default behavior to orient around these common use cases.
+
+In summary, the major changes are:
+
+- uv no longer attempts to package and install projects that do not define a `[build-system]`.
```

Comment:
Should we say "build and install" here? We don't talk about "packaging" much above anymore.

Reply:
Should we clarify in a bullet that we still install the projects dependencies?